app/assets/javascripts/components/top_app_bar.js.jsx

```diff
@@ -1,15 +1,34 @@
 var React = require('react');
+var NavMenu = require('./nav_menu');
+var Menu = require('./slide_menu');
+var MenuItem = require('./menu_item');
 
 var TopAppBar = React.createClass({
   propTypes: {
     title: React.PropTypes.string
   },
-
+  showLeft: function() {
```

Comment:
It's not really good practice to have components manipulate state directly. We should put as much of this stateful logic as we can into a separate non-React module. We can talk more about this on Wednesday.